### PR TITLE
Sign up page

### PR DESCRIPTION
**Pull request by Peteredwardj**
Resolves https://app.asana.com/0/1205761002640692/1205809435931791

- Add `/signup` route
- Add sign-up page
- Add redirect to `/signup` from login page when clicking "Don't have an account? Sign up "
- Logins and redirect to `/user` upon successful sign-up 
- Add signup unit tests

Notes: 
-  Image is from https://unsplash.com/ so it's free to use
- Not sure how it looks on bigger screens and almost sure it will get distorted on mobile (I'm assuming we're doing desktop only anyway?)

https://github.com/ECE444/Byte_Benevolence/assets/51882829/5ea4a2d3-17e5-4dff-938f-de29c07ee6b5

### CODE REVIEW

File: BBapp/templates/signup.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    <img
+      src="{{ url_for('static', filename='assets/signup_splash.jpg') }}"
+      alt="Splash Image"
+      class="img-fluid"
+      style="width: 100%; height: 950px"
+    />
+  </div>
+  <div class="col-md-2">
+    <h1 class="signup-header">Sign up</h1>
+    {%for err in errors%}
```
Comment (Peteredwardj) [inline BBapp/templates/signup.html:14]: Validation errors will show here

File: BBapp/__init__.py
Comment (lucasnchoi) [inline BBapp/__init__.py:1]: good bp

File: BBapp/forms.py
Comment (lucasnchoi) [inline BBapp/forms.py:1]: forms look good! I'll make sure to add these fields when I create the user class.

File: BBapp/routes.py
Comment (lucasnchoi) [inline BBapp/routes.py:1]: looks good!

File: BBapp/templates/login.html
Comment (lucasnchoi) [inline BBapp/templates/login.html:1]: yep makes sense to have the link on the login page.

File: requirements.txt [hunk 2]
```diff
@@ -15,4 +15,5 @@ MarkupSafe==2.1.3
 packaging==23.2
 visitor==0.1.3
 Werkzeug==3.0.0
-WTForms==3.1.0
\ No newline at end of file
+WTForms==3.1.0
+pytest==7.4.2
```
Comment (lucasnchoi) [inline requirements.txt:19]: thanks for keeping reqs up to date!

File: tests/app_test.py
Comment (lucasnchoi) [inline tests/app_test.py:1]: Good unit tests.

**Review by lucasnchoi — APPROVED** [review]
Looks good to me! Good to merge assuming you've run your own unit tests.

**Pull request merged**